lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt

```diff
@@ -45,6 +49,78 @@ class CollapsingToolbarScaffoldState(
 		get() = offsetYState.value
 
 	internal val offsetYState = mutableStateOf(initialOffsetY)
+
+	// TODO: Maybe should move toolbar expand/collapse methods to CollapsingToolbarState
+	//  but offset expand/collapse leave in CollapsingToolbarScaffoldState?
+
+	// TODO: A strange jump in snap speed is often observed
+	@ExperimentalToolbarApi
+	suspend fun expand(duration: Int = CollapsingToolbarDefaults.ExpandDuration) {
+		val anim = AnimationState(toolbarState.height.toFloat())
+
+		toolbarState.scroll {
+			var prev = anim.value
+			anim.animateTo(toolbarState.maxHeight.toFloat(), tween(duration)) {
+				scrollBy(value - prev)
+				prev = value
+			}
+		}
+	}
+
+	// TODO: A strange jump in snap speed is often observed
+	@ExperimentalToolbarApi
+	suspend fun collapse(duration: Int = CollapsingToolbarDefaults.CollapseDuration) {
+		val anim = AnimationState(toolbarState.height.toFloat())
+
+		toolbarState.scroll {
+			var prev = anim.value
+			anim.animateTo(toolbarState.minHeight.toFloat(), tween(duration)) {
+				scrollBy(value - prev)
+				prev = value
+			}
+		}
+	}
+
+	@ExperimentalToolbarApi
+	suspend fun expandOffset(snapStrategy: SnapStrategy) {
+		val anim = AnimationState(offsetYState.value.toFloat())
+
+		anim.animateTo(0f, tween(snapStrategy.expandDuration)) {
+			offsetYState.value = value.toInt()
+		}
+	}
+
+	@ExperimentalToolbarApi
+	suspend fun collapseOffset(snapStrategy: SnapStrategy) {
+		val anim = AnimationState(offsetYState.value.toFloat())
+
+		anim.animateTo(-toolbarState.minHeight.toFloat(), tween(snapStrategy.collapseDuration)) {
+			offsetYState.value = value.toInt()
+		}
+	}
+
+	// TODO: Is there a better solution rather OptIn ExperimentalToolbarApi?
+	@OptIn(ExperimentalToolbarApi::class)
+	internal suspend fun processSnap(strategy: SnapStrategy) {
+		if (toolbarState.progress > strategy.edge) {
+			expand(strategy.expandDuration)
+		} else {
+			collapse(strategy.collapseDuration)
+		}
+	}
```

Comment:
Should add a check for whether the toolbar needs to be snapped at all.. otherwise running into some weird visual bugs on horizontal scrolls inside the CollapsingToolbarScaffold content.

```suggestion
	@OptIn(ExperimentalToolbarApi::class)
	internal suspend fun processSnap(strategy: SnapStrategy) {
           if (progress != 1f && progress != 0f) {
              if (progress > strategy.edge) {
                expand(strategy.expandDuration)
            } else {
                collapse(strategy.collapseDuration)
            }
        }
	}
```

Reply:
Wonderful idea. Thank you for sharing. I think it worth to refactor `progress != 1f && progress != 0f` to remove unnecessary if-nesting. Can you do it please?

Reply:
Also you can use Kotlin range syntax for it:
```kotlin
if (progress in 0f..1f)
```

Reply:
Created pull-request, please have a look at it here: https://github.com/RareScrap/compose-collapsing-toolbar/pull/1

Reply:
Merged. Thank you for your participation.